src/blockchain/extrinsic.ts

```diff
@@ -65,7 +66,14 @@ export class Extrinsic<T extends ISubmittableResult = ISubmittableResult, C exte
   public signAndSend(nonce?: number): Promise<ParsedEventResult> {
     return firstValueFrom(
       this.extrinsic.signAndSend(this.keys, { nonce }).pipe(
-        filter(({ status }) => status.isInBlock || status.isFinalized),
+        tap(({ status }) => {
+          if (!status.isInBlock || !status.isFinalized) {
+            setTimeout(() => {
+              this.signAndSend(nonce);
+            }
+            , SECONDS_PER_BLOCK);
+          }
```

Comment:
@JoeCap08055  @wilwade  not sure if i capture your suggest correctly here

Reply:
Not sure we want to blindly re-submit... especially using SECONDS_PER_BLOCK as a timeout, in case the chain is taking longer than expected to form blocks (not unusual on Rococo...)

Reply:
tldr: I think this solution doesn't work, but coming up with a comprehensive one is not going to be straight forward.

Hmm... Side note first:
There is a [rxjs timeout](https://rxjs.dev/api/operators/timeout) or [timeoutWith](https://rxjs.dev/api/index/function/timeoutWith) that might be helpful, but you have to wrap correctly.

Rx takes a bit of work, especially when it comes to error cases and that's what's tripping you up here I think.

Few things that this would fail on I think.

- The original promise (created by the `firstValueFrom`) would hang and never resolve
- Given a bad call, this would loop ♾️ 
- We don't log or capture the errors
- There are a lot of other statuses that come through (broadcast for example) which would trigger this before the status became isInBlock or isFinalized.

Hmm... Options or some combination of them that I see:

1. Keep it how it was + pipe into a timeout/timeoutWith to error out if we don't get the status through.
2. Actually try to capture the error statuses. This example might help some, but doesn't cover some of the issues: https://polkadot.js.org/docs/api/cookbook/tx#how-do-i-get-the-decoded-enum-for-an-extrinsicfailed-event
3. Switch it up to stop using rx and instead get back the Result and use it https://github.com/polkadot-js/api/blob/master/packages/api/src/submittable/Result.ts
4. If we do any retry stuff, perhaps we need to separate out sign from send so we and resend without re-signing?

In general the retry logic is going to be iffy to put in here. We might need to start from 0 and rethink what this function is doing and what the response should be.

Reply:
> 3. Switch it up to stop using rx and instead get back the Result and use it https://github.com/polkadot-js/api/blob/master/packages/api/src/submittable/Result.ts

➕ -- I dislike RxJs anyway

I think this speaks to the fact that we might need a more general framework for interacting with the chain using PolkadotJS, since we seem to be building out more services with this technology (unless we re-examine our current super-rapid MVP approach and develop in some other stack like Java or Rust (C++, anyone? 😁 )

Reply:
I am a fan of RxJS, but it requires having the application think in it. (It is super useful for UI updates). This doesn't use it past this point.

I'm not sure what new problems we end up with however.

Reply:
I am confused here, if anyone can help me fix this

Reply:
I can pair some on it with you and we can see what ideas we can come up with.

Reply:
> I am a fan of RxJS, but it requires having the application think in it. (It is super useful for UI updates). This doesn't use it past this point.
> 
> I'm not sure what new problems we end up with however.

Clarrificatioion: not a fan of RxJS _for backend applications_. Makes much more sense for frontend/full-stack. But then you'd probably be using React or something at that point, anyway.

Reply:
incorporated rxjs timeout for better resubmission, if that will keep us afloat for reconnection-service